### ISSUE TITLE
Chapter 2 Dynamically hiding Totals Subframe example Step 5

### ISSUE THREAD

**Issue by tscottjendev** [PacktPublishing/Automated-Testing-in-Microsoft-Dynamics-365-Business-Central-Second-Edition#22]
The procedure SetShowLinesSubFrame has code that is not part of the requirements stated in the example.  Based on the premise: 
> Adjust the application code just enough to make it pass, that is, green

The requirement is: 
> When the VAT business posting group of a sales document (header) boils down to a 0% 
> VAT calculation the totals sub frame should be hidden, irrespective of the value of the VAT 
> product posting group on any of the lines.

And the test statements are:
> 1. The totals sub frame on sales invoice page is not shown when VAT business posting group 
> on the header equals a value that always leads to 0 % VAT calculation.
> 2. The totals sub frame on sales invoice page is shown when VAT business posting group on 
> the header equals a value that does not always lead to 0 % VAT calculation
> 

The current code for the setting of the variable that determines the subframe's visibility is:
```
    trigger OnAfterGetCurrRecord()
    begin
      SubFrameVisible := SetShowLinesSubFrame()
    end;

    local procedure SetShowLinesSubFrame() Result: Boolean
    var
        SalesHeader: Record "Sales Header";
        DocumentTypeFilter, DocumentNoFilter : Text;
    begin
        Rec.FilterGroup(2);
        DocumentTypeFilter := Rec.GetFilter("Document Type");
        if (DocumentTypeFilter <> '') and (Rec.GetRangeMin("Document Type") = Rec.GetRangeMax("Document Type")) then begin
            Rec.FilterGroup(4);
            DocumentNoFilter := Rec.GetFilter("Document No.");
            if (DocumentNoFilter <> '') and (Rec.GetRangeMin("Document No.") = Rec.GetRangeMax("Document No.")) then begin
                SalesHeader.Get(Rec."Document Type", Rec."Document No.");
                Result := not DoesZeroPercentVATCalculationAlwaysApply(SalesHeader."VAT Bus. Posting Group");
            end;
        end;
        Rec.FilterGroup(0);
    end;
```
The requirement (and the test statements) listed does not specify that there is any conditions under which the subframe should never be displayed (which would happen if there was no Document Type and/or Document No. filters on Rec) when testing for zero percent VAT. 
The code can be more simply expressed and still meet the requirements.  For example:
```
    trigger OnAfterGetCurrRecord()
    begin
      SubFrameVisible := SetShowLinesSubFrame()
    end;

    local procedure SetShowLinesSubFrame() Result: Boolean
    var
        SalesHeader: Record "Sales Header";
    begin
        SalesHeader.Get(Rec."Document Type", Rec."Document No.");
        exit(not DoesZeroPercentVATCalculationAlwaysApply(SalesHeader."VAT Bus. Posting Group"));
    end;
```
On initial examination, I am not even sure why the filter checking needs to be done, given the context of the subpage and Rec variable (Record Sales Line). 

In summary, to keep the example code true to the value statements being demonstrated, the code must be more succinct.   Future requirements might require additional coding or a change in requirements may make these tests obsolete but until then no extra code is required.

**Comment by lvanvugt** [comment]
Thanx, @tscottjendev, for reporting. As already discussed it seems to make sense, but there probably was a reason why I wrote this "shitty" code. "Shitty" as it is not self explanatory enough.
